openhtf/io/output/output_to_file.py

```diff
@@ -0,0 +1,32 @@
+"""Output to file.
+
+Output modules inherit from OutputToFile and overload methods
+based on output file requirements.
+
+"""
+
+import os
+import tempfile
+
+
+class OutputToFile(object):
+  """Class for saving to file."""
+
+  def Save(self):
+    """Save to file or file object."""
+    self.filename = None
```

Comment:
this class needs an `__init__`, and filename_pattern needs to be set there.  i'm guessing the multiple inheritance broke your ability to have an __init__ here, so once you fix that it should work like you expect.

base classes need to be fully functional self-contained classes, here you're assuming all sorts of behavior on the part of the subclasses (self.data and self.filename_pattern being set).  at most, a base class can leave some methods as "the subclass must implement this" (referred to as an abstract class), but making assumptions about attributes on `self` that subclasses are going to initialize/set for you makes for very fragile code.